mongodbatlas/client.go

```diff
@@ -0,0 +1,90 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"os"
+
+	"github.com/mongodb-forks/digest"
+	"github.com/turbot/steampipe-plugin-mongodbatlas/constants"
+	"github.com/turbot/steampipe-plugin-sdk/v3/plugin"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+// getMongodbAtlasClient :: returns a mongodbatlas client to perform API requests.
+func getMongodbAtlasClient(ctx context.Context, d *plugin.QueryData) (*mongodbatlas.Client, error) {
+	// Try to load client from cache
+	if cachedData, ok := d.ConnectionManager.Cache.Get(constants.CacheKeyMongodbAtlasClient); ok {
+		return cachedData.(*mongodbatlas.Client), nil
+	}
+
+	// Get mongodbatlas keys
+	publicKey, privateKey, err := getKeysFromConfig(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	// create the mongodbatlas client
+	client := createClient(ctx, publicKey, privateKey)
+
+	// save client in cache
+	d.ConnectionManager.Cache.Set(constants.CacheKeyMongodbAtlasClient, client)
+
+	return client, nil
+}
+
+// getKeysFromConfig fetches the public and private keys from the connection config
+// falls back to the environment variables if it cannot find one in the config
+// returns an error if both keys could not be resolved
+func getKeysFromConfig(ctx context.Context, d *plugin.QueryData) (publicKey string, privateKey string, _ error) {
+	config := GetConfig(d.Connection)
+
+	// Get the authorization publicKey
+	publicKey = os.Getenv(constants.EnvKeyClientPublicKey)
```

Comment:
The Terraform provider accepts the `MONGODB_ATLAS_PUBLIC_KEY` or `MCLI_PUBLIC_API_KEY` env variable for this (and similarly 2 different env vars for the private key). Does the MongoDB Atlas CLI support both of these environment variables as well? If so, we should probably support both of these as well.